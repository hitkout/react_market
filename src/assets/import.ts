import ComputerFor1Card from './images/computer1.png';
import ComputerFor2Card from './images/computer2.png';
import ComputerFor3Card from './images/computer3.png';
import ComputerFor4Card from './images/computer4.png';
import ComputerFor5Card from './images/computer5.png';
import ComputerFor6Card from './images/computer6.png';
import ComputerFor7Card from './images/computer7.png';
import ComputerFor8Card from './images/computer8.png';
import FileIcon from './images/file.svg';
import UserIcon from './images/user.svg';
import PercentIcon from './images/percent.svg';
import RuSvg from './images/ru.svg';
import EnSvg from './images/en.svg';

export {
    ComputerFor1Card,
    ComputerFor2Card,
    ComputerFor3Card,
    ComputerFor4Card,
    ComputerFor5Card,
    ComputerFor6Card,
    ComputerFor7Card,
    ComputerFor8Card,
    FileIcon,
    UserIcon,
    PercentIcon,
    RuSvg,
    EnSvg
}