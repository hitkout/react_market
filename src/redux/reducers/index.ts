import { combineReducers } from 'redux';

import productsReducer from './products';
import basketReducer from './basket';
import advantagesReducer from './advantages';
import productPageReducer from './productPage';

const rootReducer = combineReducers({
    products: productsReducer,
    basket: basketReducer,
    advantages: advantagesReducer,
    productPage: productPageReducer
})

export default rootReducer;