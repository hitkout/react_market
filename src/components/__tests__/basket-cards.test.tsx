import React from 'react';
import { mount } from 'enzyme';
import BasketCards from '../basket-cards';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import { StaticRouter } from 'react-router-dom';
import { describe, it, expect } from '@jest/globals';
import { ComputerFor1Card } from '../../assets/import';

describe('Тестирование Basket', () => {

    it('Тестируем рендер Basket', async () => {
        const component = mount(
            <Provider store={store}>
                <StaticRouter>
                    <BasketCards id={5} image={ComputerFor1Card} name="Computer" totalPrice={40000} totalCount={5} onRemove onMinus onPlus />
                </StaticRouter>
            </Provider>
        )

        expect(component).toMatchSnapshot()
    });
});