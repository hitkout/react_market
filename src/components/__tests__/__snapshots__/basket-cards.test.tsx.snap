// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование Basket Тестируем рендер Basket 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <StaticRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      staticContext={Object {}}
    >
      <BasketCards
        id={5}
        image="FILE_MOCK"
        name="Computer"
        onMinus={true}
        onPlus={true}
        onRemove={true}
        totalCount={5}
        totalPrice={40000}
      >
        <div
          className="CatalogCard"
        >
          <div
            className="cardRow"
          >
            <div
              className="catalogBlockImage"
            >
              <img
                alt="Computer"
                className="catalogCardImage"
              />
            </div>
            <p
              className="catalogCardTitle"
            >
              Computer
            </p>
            <div
              className="counterStyle"
            >
              <button
                className="buttonStyleBasketIcon"
                onClick={[Function]}
              >
                <i>
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f056",
                          "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z",
                        ],
                        "iconName": "minus-circle",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-minus-circle fa-w-16 "
                      data-icon="minus-circle"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zM124 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H124z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </i>
              </button>
              5
              <button
                className="buttonStyleBasketIcon"
                onClick={[Function]}
              >
                <i>
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f055",
                          "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z",
                        ],
                        "iconName": "plus-circle",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-plus-circle fa-w-16 "
                      data-icon="plus-circle"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </i>
              </button>
            </div>
            <div
              className="priceProduct"
            >
              40000
              <i>
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        384,
                        512,
                        Array [],
                        "f158",
                        "M239.36 320C324.48 320 384 260.542 384 175.071S324.48 32 239.36 32H76c-6.627 0-12 5.373-12 12v206.632H12c-6.627 0-12 5.373-12 12V308c0 6.627 5.373 12 12 12h52v32H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v52c0 6.627 5.373 12 12 12h58.56c6.627 0 12-5.373 12-12v-52H308c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12H146.56v-32h92.8zm-92.8-219.252h78.72c46.72 0 74.88 29.11 74.88 74.323 0 45.832-28.16 75.561-76.16 75.561h-77.44V100.748z",
                      ],
                      "iconName": "ruble-sign",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-ruble-sign fa-w-12 "
                    data-icon="ruble-sign"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M239.36 320C324.48 320 384 260.542 384 175.071S324.48 32 239.36 32H76c-6.627 0-12 5.373-12 12v206.632H12c-6.627 0-12 5.373-12 12V308c0 6.627 5.373 12 12 12h52v32H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h52v52c0 6.627 5.373 12 12 12h58.56c6.627 0 12-5.373 12-12v-52H308c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12H146.56v-32h92.8zm-92.8-219.252h78.72c46.72 0 74.88 29.11 74.88 74.323 0 45.832-28.16 75.561-76.16 75.561h-77.44V100.748z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </i>
            </div>
            <div
              className="crossIcon"
            >
              <button
                className="buttonStyleBasketIcon"
                onClick={[Function]}
              >
                <i>
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f057",
                          "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
                        ],
                        "iconName": "times-circle",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-times-circle fa-w-16 "
                      data-icon="times-circle"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </i>
              </button>
            </div>
          </div>
        </div>
      </BasketCards>
    </Router>
  </StaticRouter>
</Provider>
`;
